### ISSUE TITLE
Documentation for how pytorch tensor.view() rearranges dimensions?

### ISSUE THREAD

**Issue by mhgump** [pytorch/pytorch#2013]
I can't seem to find any documentation about the tensor.view() function or how it is implemented. I am trying to verify that pytorch view will always consistently reshape dimensions. Specifically I am trying to apply the softmax function onto a 4D tensor. I want to be able to verify that

'''
tensor = tensor.view(-1,N)
tensor = nn.Softmax()(tensor)
tensor = tensor.view(-1,N,spatial_dim_1,spatial_dim_2)
'''

will preserve the spatial orientation of the tensor's cells.

Thanks!

EDIT: Verified it myself

**Comment by fmassa** [comment]
@mhgump `torch.view` only changes the sizes of the tensor, while the underlying content remains the same, pretty much like numpy reshape, so the order is the same as the order in the underlying tensor.
As an example, say we have a tensor `x` of size `5x4x3x2`. Doing `x.view(5, -1)`  just rearranges the sizes (you can see by creating a `torch.arange(0, 5 * 4 * 3 * 2)` tensor, view it and view back).

For the source code, it was moved to `C` for efficiency, but you can find the old python implementation [here](https://github.com/pytorch/pytorch/blob/518864a7e094f32044ef4c0de82c0f20f4ed99c2/torch/tensor.py#L178-L217).

One more note, if your tensor is of shape `N x C x H x W`, and you want to apply softmax on `C`, you need to transpose it before viewing it. Something like

```python
a = torch.rand(N, C, H, W)
a_t = a.permute(0, 2, 3, 1).contiguous().view(-1, C)
# ... perform softmax
# view back
res = res_t.view(N, H, W, C).permute(0, 3, 1, 2)
```

**Comment by mhgump** [comment]
@fmassa Thanks for the notes, I ended up permuting the axes like you suggested

**Comment by hmishfaq** [comment]
Hey, @fmassa and @mhgump, I am in somewhat similar situation and trying to verify that the spatial orientation is preserved. 

So, in my case I am sequence of video frames and thus say size of the tensor is ```B x S x C x H x W``` where B denotes batch size and S denotes length of each sequence considered. 

So what I want to do is to pass each frame through an encoder (for simplicity let's say just one ```Conv2d``` layer). For this I am considering something along this line
```

a = torch.rand(B, S, C, H, W)
a = torch.view(-1,C,H,W)
encoder = nn.Conv2d(C, C_out, 3, stride=2)
a = encoder(a)
_, C_out, H_out, W_out = a.size()
a = a.view(B, S, C_out, H_out, W_out)
```

Will it preserve the spatial order as in the original ```a```

**Comment by dizcza** [comment]
I cannot find `torch.flatten` in v0.4.1 docs either. Does it copy the output tensor? I suppose not.

**Comment by wxyhv** [comment]
@fmassa Hello, you said "For the source code, it was moved to C for efficiency, but you can find the old python implementation here.". Could you tell me where is the C code for view? I want to study how to implement the "view" operation in C. 
Dose this cpp file contain the C code for view? aten/src/ATen/native/TensorShape.cpp
Thank you Very much.
